Text Generation WebUI/scripts/generate-config-schema-from-csv.cjs

```diff
@@ -0,0 +1,298 @@
+'use strict';
+const assert = require('assert/strict');
+const fs = require('fs');
+const { type } = require('os');
+
+const csvFilePath = process.argv[2];
+if (!csvFilePath) {
+  throw 'Specify csv file path as parameter';
+}
+
+const headersObj = {
+  category: 'Category',
+  group: 'Group',
+  subgroup: 'Subgroup',
+  name: 'Name',
+  variable: 'Variable',
+  cmdParameter: 'cmd parameter',
+  type: 'Type',
+  valueType: 'ValueType',
+  options: 'Options',
+  useConditions: 'Use Condition',
+  default: 'Default Value',
+  min: 'Min Value',
+  max: 'Max Value',
+  description: 'Description',
+};
+
+const types = {
+  category: 0,
+  group: 1,
+  subgroup: 2,
+  number: 3,
+  text: 4,
+  textarea: 5,
+  checkbox: 6,
+  radio: 7,
+  select: 8,
+  multiselect: 9,
+  slider: 10,
+};
+
+const valueType = {
+  default_type: 0,
+  double: 1,
+  float: 2,
+  int32: 3,
+  int64: 4,
+  uint32: 5,
+  uint64: 6,
+  sint32: 7,
+  sint64: 8,
+  fixed32: 9,
+  fixed64: 10,
+  sfixed32: 11,
+  sfixed64: 12,
+  bool: 13,
+  string: 14,
+  bytes: 15,
+};
+
+const splitByComma = (str) => {
+  const result = [];
+  let current = '';
+  let insideQuotes = false;
+
+  for (let i = 0; i < str.length; i++) {
+    const char = str[i];
+
+    if (char === '"') {
+      insideQuotes = !insideQuotes; // Toggle the insideQuotes flag
+    }
+
+    if (char === ',' && !insideQuotes) {
+      result.push(current.trim());
+      current = '';
+    } else if (char !== '"') {
+      current += char;
+    }
+  }
```

Comment:
```suggestion
const splitByComma = (str) => {
  const regex = /(?:[^,"\\]+|"(?:\\.|[^"\\])*")+/g;
  const result = str.match(regex) || [];
  return result.map(item => item.trim().replace(/^"(.*)"$/, '$1'));
};
```

Reply:
@marchuk-vlad 
I missed that you were also returning empty values. With this code, it works:

```typescript
const splitByComma = (str) => {
    const regex = /(?:[^,"\\]+|"(?:\\.|[^"\\])*"|)(?=,|$)/g;
    const result = str.match(regex) || [];
    return result.map(item => item.trim().replace(/^"(.*)"$/, '$1'));
};
```

Reply:
@marchuk-vlad, It looks like we have the wrong order of fields after the regex. Okay, I’ll try to figure it out later.